### PR TITLE
Query: Convert non-context related property accesses to constant in q…

### LINKED ISSUE #10271: 2.0.1-rtm-203 regression: System.InvalidCastException: Unable to cast object of type 'System.Int32' to type 'System.Linq.Expressions.LambdaExpression'
When configuring a query filter that captures a variable (I think that's whats it called?) in the expression 2.0.1-rtm-203 throws an `InvalidCastException`. This doesn't happen in 2.0.0

``` c#
// Works
protected override void OnModelCreating(ModelBuilder modelBuilder)
{
    modelBuilder.Entity<MyEntity>().HasQueryFilter(e => e.TenantId == 123);
}

// Throws InvalidCastException when executing a query.
protected override void OnModelCreating(ModelBuilder modelBuilder)
{
    int tenantId = 123;
    modelBuilder.Entity<MyEntity>().HasQueryFilter(e => e.TenantId == tenantId);
}
```
**OR**
1. `git clone https://github.com/nphmuller/EfCoreQueryFilterBug`
2. `dotnet run` throws exception. (See below for details)
3. open .csproj file and change version from 2.0.1-* to 2.0.0
4. `dotnet run` completes succesfully.

Same thing occurs when I use:
`https://dotnet.myget.org/gallery/aspnet-2-0-2-october2017-patch` - 2.0.1-rtm-207
instead of:
`https://dotnet.myget.org/gallery/aspnet-2-0-2-october2017-patch-public` - 2.0.1-rtm-203

Exception details:
```
Unhandled Exception: System.AggregateException: One or more errors occurred. (Unable to cast object of type 'System.Int32' to type 'System.Linq.Expressions.LambdaExpression'.) ---> System.InvalidCastException: Unable to cast object of type 'System.Int32' to type 'System.Linq.Expressions.LambdaExpression'.
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.CreateSetFilterParametersExpressions(ParameterExpression& contextVariableExpression)
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.CreateExecutorLambda[TResults]()
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.CreateQueryExecutor[TResult](QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Storage.Internal.InMemoryDatabase.CompileAsyncQuery[TResult](QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass24_0`1.<CompileAsyncQuery>b__0()
   at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQueryCore[TFunc](Object cacheKey, Func`1 compiler)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.ExecuteAsync[TResult](Expression query)
   at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryable`1.System.Collections.Generic.IAsyncEnumerable<TResult>.GetEnumerator()
   at System.Linq.AsyncEnumerable.<Aggregate_>d__6`3.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at ConsoleApp3.Program.Main(String[] args) in C:\Dev\_EfCoreQueryFilterBug\ConsoleApp3\Program.cs:line 14
```

### PR DESCRIPTION
**Pull request by smitpatel**
…uery filters

Resolves #10271

### CODE REVIEW

File: src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs [hunk 1]
```diff
@@ -364,7 +364,14 @@ private Expression TryExtractParameter(Expression expression)
...
+            if (_generateContextAccessors)
+            {
+                if (!(parameterValue is LambdaExpression))
+                {
+                    return Expression.Constant(parameterValue, expression.Type);
+                }
+            }
+            else
             {
                 if (parameterValue is Expression valueExpression)
```
Comment (anpete) [inline src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs]: Do we need to hit this branch when _generateContextAccessors == true and parameterValue is not a LambdaExpression?

**Comment by smitpatel** [comment]
Updated.

**Comment by smitpatel** [comment]
@anpete - Ping!

File: src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs [hunk 2]
```diff
@@ -21,6 +21,7 @@ public class ParameterExtractingExpressionVisitor : ExpressionVisitor
     {
         private static readonly TypeInfo _queryableTypeInfo = typeof(IQueryable).GetTypeInfo();
         private static ContextParameterReplacingExpressionVisitor _contextParameterReplacingExpressionVisitor;
+        private static readonly string _queryFilterPrefix = "ef_filter";
```
Comment (anpete) [inline src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs]: Constant in Pascal case

Comment (smitpatel) [reply src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs]: Lack of R#

File: test/EFCore.SqlServer.FunctionalTests/Query/QueryBugsTest.cs [hunk 3]
```diff
@@ -3126,6 +3126,141 @@ public class Configuration9468
...
+                    AssertSql(
+                        @"SELECT [p].[Id], [p].[TenantId]
+FROM [Posts] AS [p]
+WHERE [p].[TenantId] = 1");
+                }
+            }
+        }
+
+        [Fact]
+        public virtual void ezpz()
```
Comment (smitpatel) [inline test/EFCore.SqlServer.FunctionalTests/Query/QueryBugsTest.cs]: @maumar :trollface:

**Pull request merged**